src/Scripting/Core/ScriptBuilder.cs

```diff
@@ -73,20 +73,13 @@ public int GenerateSubmissionId(out string assemblyName, out string typeName)
             {
                 // get compilation diagnostics first.
                 diagnostics.AddRange(compilation.GetParseDiagnostics());
-                if (diagnostics.HasAnyErrors())
-                {
-                    CompilationError(diagnostics, compiler.DiagnosticFormatter);
-                }
-
+                ThrowIfAnyCompilationErrors(diagnostics, compiler.DiagnosticFormatter);
                 diagnostics.Clear();
```

Comment:
We were only reporting warnings previously if there were errors as well. Is that the desired behavior?